### PR TITLE
Pushdown domains with many ranges in Oracle connector

### PR DESCRIPTION
**Pull request by losipiuk**
If domain has more that 1000 ranges it is simplified, before pushdown
attempt. Without this query would fails with:
ORA-01795: maximum number of expressions in a list is 1000

### CODE REVIEW

File: presto-oracle/src/test/java/io/prestosql/plugin/oracle/TestOracleDistributedQueries.java [hunk 1]
```diff
@@ -128,7 +128,7 @@ protected TestTable createTableWithDefaultColumns()
     @Override
     public void testLargeIn()
     {
-        int numberOfElements = 1000;
+        int numberOfElements = 5000;
```
[the remark this answers is not in the document]
Comment (losipiuk) [reply presto-oracle/src/test/java/io/prestosql/plugin/oracle/TestOracleDistributedQueries.java:131]: There is still this extra part in Oracle test which is not present in superclass:
```
        String arrayValues = range(0, numberOfElements)
                .mapToObj(i -> format("ARRAY[%s, %s, %s]", i, i + 1, i + 2))
                .collect(joining(", "));
        assertQuery("SELECT ARRAY[0, 0, 0] in (ARRAY[0, 0, 0], " + arrayValues + ")", "values true");
        assertQuery("SELECT ARRAY[0, 0, 0] in (" + arrayValues + ")", "values false");
```

I can call out to super anyway though, even if we still want test with arrays.

[the remark this answers is not in the document]
Comment (eskabetxe) [reply presto-oracle/src/test/java/io/prestosql/plugin/oracle/TestOracleDistributedQueries.java:131]: @losipiuk that code isnt oracle especific, that method have that code before..
that was removed here https://github.com/prestosql/presto/commit/0c13d169dcf189d4a913187de6e33c212df92bfa

File: presto-oracle/src/main/java/io/prestosql/plugin/oracle/OracleClient.java [hunk 2]
```diff
@@ -332,6 +345,19 @@ else if (precision > Decimals.MAX_PRECISION || columnSize <= 0) {
...
+    private static PredicatePushdownController.DomainPushdownResult fullPushdownIfSupported(Domain domain)
+    {
+        if (domain.getValues().getRanges().getRangeCount() > ORACLE_MAX_LIST_EXPRESSIONS) {
+            // pushdown simplified domain
+            Domain pushedDown = domain.simplify();
+            return new PredicatePushdownController.DomainPushdownResult(pushedDown, domain);
+        }
+        else {
+            // full pushdown
+            return new PredicatePushdownController.DomainPushdownResult(domain, Domain.all(domain.getType()));
```
[the remark this answers is not in the document]
Comment (losipiuk) [reply presto-oracle/src/main/java/io/prestosql/plugin/oracle/OracleClient.java:357]: Added.

**Comment by losipiuk** [comment]
ac

**Comment by losipiuk** [comment]
#4885

**Pull request merged**